src/nvim/message.c

```diff
@@ -1450,6 +1450,9 @@ void msg_prt_line(char_u *s, int list)
       } else if (c == ' ' && trail != NULL && s > trail) {
         c = lcs_trail;
         attr = hl_attr(HLF_8);
+      } else if (c == ' ' && list && lcs_space != NUL) {
```

Comment:
@Pyrohh Thanks for the pointer!

When in doubt, I look to Tcl and Lua for exemplary C code (they use `'\0'`). But the point is taken.